stellargraph/core/element_data.py

```diff
@@ -303,6 +305,37 @@ def _numpyise(d, dtype):
     return {k: np.array(v, dtype=dtype) for k, v in d.items()}
 
 
+class FlatAdjacencyList:
+    def __init__(self, flat_array, neighbor_counts):
+        self.splits = np.cumsum(np.append(0, neighbor_counts))
+        self.max_node_iloc = len(neighbor_counts) - 1
+        self.flat = flat_array
+        self.empty = np.array([], dtype=np.uint8)
+
+    def __getitem__(self, idx):
+        start, stop = self.splits[idx], self.splits[idx + 1]
+        return self.flat[start:stop]
+
+    def get(self, idx, default):
+        if idx > self.max_node_iloc:
```

Comment:
This case should never happen now, and we should use `[]` instead of `get` in `.edge_ilocs`. In particular, previously isolated nodes would not appear in the dictionary, but now all nodes appear in `splits`.

That is, we can delete this.

Reply:
confusingly `self.max_node_iloc` isn't always the max node iloc, its the max node iloc in the edges. If the max node iloc is isolated then it will be larger than `self.max_node_iloc` and so we should check for this

Reply:
there's a brief coment in `init_directed_adj` about this but I'll write another comment here  👍

Reply:
I'll need to pass `max_node_iloc` into the constructor here but I think its worth it. 👍 

We should also be able to get the node degrees rapidly using `np.diff(self._edges_dict.splits)`!